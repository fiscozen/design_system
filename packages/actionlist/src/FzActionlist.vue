<template>
  <div class="fz__actionlist bg-core-white inline-flex grow-0 flex-col rounded p-4">
    <div v-if="label" class="text-grey-400 flex h-32 items-center px-12 text-xs">
      <span>{{ label }}</span>
    </div>
    <div class="flex flex-col" v-for="(item, itemIndex) in items" :key="itemIndex">
      <FzNavlink
        v-if="item.type === 'button'"
        class="grow-1 flex justify-start"
        v-bind="item"
        @click="emit('fzaction:click', itemIndex, item)"
        >{{ item.label }}</FzNavlink
      >
      <FzRouterNavlink
        v-else
        class="grow-1 flex justify-start"
        v-bind="item"
        @click="emit('fzaction:click', itemIndex, item)"
        >{{ item.label }}</FzRouterNavlink
      >
    </div>
  </div>
</template>

<script lang="ts" setup>
import { FzNavlink, FzRouterNavlink } from '@fiscozen/navlink'
import { FzActionlistProps, ActionlistItem } from './types'

const props = defineProps<FzActionlistProps>()
const emit = defineEmits<{
  'fzaction:click': [index: number, action: ActionlistItem]
}>()
</script>

<style>
.fz__actionlist {
  min-width: 240px;
  box-shadow:
    0px 1px 3px 0px rgba(0, 0, 0, 0.1),
    0px 1px 2px 0px rgba(0, 0, 0, 0.06);
}
</style>
