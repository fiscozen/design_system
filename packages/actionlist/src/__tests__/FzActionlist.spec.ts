import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import FzActionlist from '../FzActionlist.vue'

describe('FzActionlist', () => {
  it('should match snapshot', () => {
    const wrapper = mount(FzActionlist, {
      props: {
        label: 'This is a label',
        items: [
          {
            label: 'Item #1',
            disabled: true,
            meta: {
              path: '/foo',
              name: 'foo'
            }
          },
          {
            label: 'Item #2',
            meta: {
              path: '/foo',
              name: 'foo'
            }
          }
        ]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
