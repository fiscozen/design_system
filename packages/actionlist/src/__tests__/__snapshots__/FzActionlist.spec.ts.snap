// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzActionlist > should match snapshot 1`] = `
"<div class="fz__actionlist inline-flex grow-0 flex-col rounded p-4">
  <div class="text-grey-400 flex h-32 items-center px-12 text-xs"><span>This is a label</span></div>
  <div class="flex flex-col">
    <router-link to="[object Object]" disabled="true" class="px-12 py-6 text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent font-medium grow-1 flex justify-start">
      <!--v-if-->Item #1
    </router-link>
  </div>
  <div class="flex flex-col">
    <router-link to="[object Object]" disabled="false" class="px-12 py-6 text-grey-500 hover:bg-background-alice-blue border-1 focus:border-1 focus:bg-background-alice-blue disabled:text-grey-100 disabled:bg-core-white h-32 items-center rounded border-transparent text-sm hover:text-blue-500 focus:border-blue-500 focus:text-blue-500 disabled:border-transparent font-medium grow-1 flex justify-start">
      <!--v-if-->Item #2
    </router-link>
  </div>
</div>"
`;
